spec/modules.md

```diff
@@ -0,0 +1,428 @@
+    partials][resolving for partials].
+
+  * If neither `sass` nor `scss` are null, throw an error.
+
+  * Otherwise, if exactly one of `sass` and `scss` is null, return the other
+    one.
+
+  * Otherwise, if the result of [resolving `url` + `".import.css"` for
+    partials][resolving for partials] is not null, return it.
+
+* Otherwise, let `sass` be the result of [resolving `url` + `".sass"` for
+  partials][resolving for partials].
+
+* Let `scss` be the result of [resolving `url` + `".scss"` for
+  partials][resolving for partials].
+
+* If neither `sass` nor `scss` are null, throw an error.
+
+* Otherwise, if exactly one of `sass` and `scss` is null, return the other
+  one.
+
+* Otherwise, return the result of [resolving `url` + `".css"` for
+  partials][resolving for partials]. .
+
+[resolving for partials]: #resolving-a-file-url-for-partials
+
+### Resolving a `file:` URL for Partials
+
+This algorithm takes a URL, `url`, whose scheme must be `file` and returns
+either another URL that's guaranteed to point to a file on disk or null.
+
+* If `url`'s [basename](#basename) begins with `"_"`:
+
+  * If a file exists on disk at `url`, return `url`.
+
+    Otherwise return null.
+
+* Let `partial` be [`dirname(url)`](#dirname) + `"_"` +
+  [`basename(url)`](#basename).
+
+* If a file exists on disk at both `url` and `partial`, throw an error.
+
+* If a file exists on disk at `url`, return `url`.
+
+* If a file exists on disk at `partial`, return `partial`.
+
+* Return null.
+
+### Resolving a Member
+
+This algorithm takes a [member](#member) name `name` and a member type `type`,
+and returns a member of type `type` or null.
+
+* If `name` is a plain `Identifier` or a `Variable` that's not a
+  `NamespacedVariable`:
+
+  * Let `scope` be the [scope][] of the innermost block containing the current
+    statement such that `scope` has a member of type `type` named `name`, or
+    null if no such scope exists.
+
+    [scope]: variables.md#scope
+
+  * If `scope` is not null, return `scope`'s value of type `type` named `name`.
+
+* If `name` is a [`NamespacedIdentifier`](#syntax) of the form
+  `namespace.raw-name` or a [`Variable`][] of the form `namespace.$raw-name`:
+
+  [`Variable`]: variables.md#syntax
+
+  * Let `use` be the [`@use` rule][] in [the current source file][] whose
+    namespace is `namespace`. If there isn't exactly one such rule, throw an
+    error.
+
+    [`@use` rule]: at-rules/use.md
+
+    > Unlike other identifiers in Sass, module namespaces *do not* treat `-` and
+    > `_` as equivalent.
+
+  * If `use` hasn't been executed yet, throw an error.
+
+  * Otherwise, let `module` be [`use`'s module][].
+
+    [`use`'s module]: at-rules/use.md#a-use-rules-module
+
+  * Return the member of `module` with type `type` and name `raw-name`. If there
+    is no such member, throw an error.
+
+* If `type` is not "variable" and the current source file contains a top-level
+  definition of a member of type `type` named `name`:
```

Comment:
I'm not really following what this line is saying, especially in relation to the note that follows. Is there an example that illustrates what this section is for?

Reply:
Added an explanation